### PR TITLE
src: Add a `server_pid` default environment variable

### PR DESCRIPTION
**Pull request by lenormf**
Hi,

Since I don't use process IDs as session names, I'd like to have a simple way of gettting the process ID of the server.

Also this feature allows making use of a feature of LSP, namely allowing the language server to have the same lifespan as the process that spawns it[1].

[1] https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#initialize-request

### CODE REVIEW

File: src/main.cc [hunk 1]
```diff
@@ -157,6 +157,10 @@ void register_env_vars()
             "window_height", false,
             [](StringView name, const Context& context) -> String
             { return to_string(context.window().dimensions().line); }
+        }, {
+            "server_pid", false,
+            [](StringView name, const Context& context) -> String
+            { return to_string(Server::instance().pid()); }
```
Comment (mawww) [inline src/main.cc]: Or we could just call `getpid` here and not bother storing that in the `Server`

Comment (lenormf) [inline src/main.cc]: It would issue a syscall everytime the variable is expanded, since it's not going to change I think storing it is cleaner.

**Comment by mawww** [comment]
This is going to be `$PPID` in any `%sh{...}` block, so we dont need to expose it.

**Comment by lenormf** [comment]
You still need to spawn a shell, does it not make sense to have a dedicated option accessible with a simple expand?

**Comment by mawww** [comment]
I find it highly unlikely that you need the server pid, but are not forking a shell to do something with it at some point.

**Comment by lenormf** [comment]
Killing the server from outside a renamed session.

**Comment by mawww** [comment]
Not sure i get you, what command would you use ?

**Comment by lenormf** [comment]
I wanted to be able to run `kak -d -s my-session`, and list information about sessions (in a script, looping over `kak -l`), such as server PID, number of open buffers etc from the CLI. To do that, I wanted to introduce an `-F` flag that would act like `tmux` formats (mentioned [in this issue](https://github.com/mawww/kakoune/issues/538#issuecomment-223156458)), and make the server expand a string passed to it (`echo '%val{server_pid}' | kak -s my-session -F`). That way I could easily kill a session from the command line, and/or make a script that lists information as requested in the issue linked previously.

Arguably this `server_pid` variable isn't really useful in itself, but I thought that if I was to implement an `-F` flag, I wouldn't want the server to have to spawn a shell or issue a syscall to get such basic info.

**Comment by lenormf** [comment]
Actually my implementation would use `-P`, which would be like `-p` but would write the expanded string passed on `stdin` on `stdout`.

**Comment by mawww** [comment]
I am still unconvinced, if the point of having the server_pid is to facilitate killing a session, might I suggest just using `echo 'kill!' | kak -p <session>` ? the pid of the server is really just an implementation detail, I am not sure how relevant it is to make it available.

**Pull request closed without merging**